src/web100srv.c

```diff
@@ -1368,28 +1381,31 @@ void process_parent_message(int parent_message, Connection *ctl,
     case SRV_QUEUE_SERVER_BUSY_60s:
       // The SRV_QUEUE_SERVER_BUSY_60s message is not emitted by any
       // codepaths in the server.  This message is likely obsolete.
-      // Set the watchdog alarm for 60 seconds, plus some slop.
+      // Set the watchdog alarm for 60 seconds, plus some slop, to prevent
+      // spurious calls to cleanup().
       alarm(120);
       break;
     default:
       // If the message was not one of the magic values, then it is a number
       // of minutes to wait until the test starts.  Set the watchdog alarm
       // for that many minutes, plus a little bit extra to avoid race
-      // conditions.
+      // conditions and prevent spurious calls to cleanup().
       alarm((parent_message + 1) * 60);
       break;
   }
 }
 
 /**
  * The code run by the child process. This function never returns, it only
- * calls exit().  It also has an alarm() timer which by default prevents any
- * client from living for more than 5 minutes.  The parent is in control, and
- * will send this child a signal when it gets to the head of the testing queue
- * (and will also send other status messages that are relayed by the client).
- * Only the child process can communicate to the client (OpenSSL connections
- * can only be used by one process), so we pass along any queueing messages we
- * get from the parent.
+ * calls exit().  It also has an alarm() timer which functions as a watchdog,
+ * and all communication sockets should have socket timeouts set.
+ *
+ * The parent process is in control, and will send this child a message on the
```

Comment:
BTW - thought for the future - why doesn't the parent kill the children at an appropriate time?